src/main/java/controller/LadderGameController.java

```diff
@@ -0,0 +1,73 @@
+package controller;
+
+import domain.ladderGame.GameInit;
+import domain.ladderGame.GameResult;
+import domain.ladderGame.LadderGame;
+import utils.Log;
+import view.InputView;
+import view.OutputView;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class LadderGameController {
+
+    private final InputView inputView = new InputView();
+    private final OutputView outputView = new OutputView();
+
+    public void run() {
+        InitLadderGame initLadderGame = new InitLadderGame(inputView);
+        GameInit gameInit = initLadderGame.getGameInit();
+        showInitResult(gameInit);
+
+        LadderGame ladderGame = new LadderGame(gameInit);
+        playGame(ladderGame);
+    }
+
+    private void showInitResult(GameInit gameInit) {
+        outputView.showInitResult(gameInit);
+    }
+
+    private void playGame(LadderGame ladderGame) {
+        GameResult gameResult = ladderGame.play();
+        showGameResult(gameResult);
+    }
+
+    private void showGameResult(GameResult gameResult) {
+        List<String> playersName = new ArrayList<>(gameResult.getResults().keySet());
+        Set<String> checkedPlayer = new HashSet<>();
+        while(!gottenAllResult(checkedPlayer, playersName)) {
+            String target = readValidatedTarget(playersName);
+            outputView.showTargetResult(target, gameResult);
+            checkedPlayer.add(target);
+        }
+        outputView.showAllResult(gameResult);
+    }
+
+    private boolean gottenAllResult(Set<String> checkedPlayer, List<String> playersName) {
+        if (checkedPlayer.containsAll(playersName))
+            return true;
+        if (checkedPlayer.contains("all"))
+            return true;
+        return false;
+    }
```

Comment:
<img width="653" alt="image" src="https://user-images.githubusercontent.com/31095063/221619041-157908e9-8a39-4891-a695-2452f039cf2e.png">

IDE는 위와 같이 추천해주네요 ㅎㅎ 

```suggestion
   private boolean gottenAllResult(Set<String> checkedPlayer, List<String> playersName) {
        return checkedPlayer.containsAll(playersName) || checkedPlayer.contains("all");
    }
```

혹은 이렇게도 표현할 수 있을 것 같아요 :)

Reply:
`checkedPlayer.containsAll(playersName)`은 모든 참여자를 조회했는지에 대한 로직이고
`checkedPlayer.contains("all")`는 조회할 참여자를 입력하라고 했을 때 "all"을 입력했는지 판단하는 로직이라
두 로직이 다른 로직이라고 생각해서 분리했었습니다.